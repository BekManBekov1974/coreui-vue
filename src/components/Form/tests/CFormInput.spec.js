import { mount } from '@vue/test-utils'
import Component from '../CFormInput'

const ComponentName = 'CFormInput'
const customWrapper = mount(Component, {
  propsData: {
    type: 'number',
    label: 'label',
    id: 'some_id',
    wasValidated: true,
    checked: true,
    value: 'value',
    validFeedback: 'input is valid',
    invalidFeedback: 'input is invalid',
    tooltipFeedback: true,
    description: 'permission',
    isValid: true,
    addInputClasses: 'additional-input-class',
    addLabelClasses: 'additional-label-class',
    addWrapperClasses: 'additional-wrapper-class',
    custom: true,
    size: 'lg'
  }
})

describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(customWrapper.element).toMatchSnapshot()
  })
})
