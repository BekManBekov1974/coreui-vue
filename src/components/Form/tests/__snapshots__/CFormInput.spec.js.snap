// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CFormInput renders correctly 1`] = `
<div
  class="c-form-group c-was-validated"
  role="group"
>
  <label
    class="additional-label-class c-col-form-label-lg"
    for="some_id"
  >
    
        label
      
  </label>
   
  <div
    class="additional-wrapper-class c-input-group c-input-group-lg"
  >
    <!---->
     
    <input
      class="c-form-control c-is-valid additional-input-class"
      custom="true"
      id="some_id"
      type="number"
    />
      
    <!---->
     
    <div
      class="c-valid-tooltip"
    >
      input is valid
    </div>
     
    <div
      class="c-invalid-tooltip"
    >
      input is invalid
    </div>
     
    <small
      class="c-form-text c-text-muted c-w-100"
    >
      permission
    </small>
  </div>
</div>
`;
